From 9468a0ea09565674e125c6148c39c44e48ee5389 Mon Sep 17 00:00:00 2001
From: Huang Chaofan <chaofan.huang@nxp.com>
Date: Thu, 12 Apr 2018 13:21:52 +0800
Subject: [PATCH 3594/5242] MLK-18006[IMX8QXP B0] VPU fault when use the
 latest toolchain for imx8qxp B0 board

commit  7e2dce6ba8e0eceb6c72ceb7555fda4e28f90ef2 from
https://source.codeaurora.org/external/imx/linux-imx.git

VPU fault when use the latest toolchain for imx8qxp B0 board, replace
"ioreamp" with "ioremap_wc" in decoder and encoder for the following
memcpy action

Signed-off-by: Huang Chaofan <chaofan.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-decoder-b0/vpu_b0.c         |    6 +++---
 drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c |    6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.c b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
index bc05382..5559277 100755
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.c
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
@@ -2061,7 +2061,7 @@ static int boot_CM4_up(struct vpu_dev *dev, void *boot_addr)
 	aux_core_ram = 0x34FE0000;
 	size = SZ_128K;
 
-	core_ram_vir = ioremap(aux_core_ram,
+	core_ram_vir = ioremap_wc(aux_core_ram,
 			size
 			);
 	if (!core_ram_vir)
@@ -2642,7 +2642,7 @@ static int vpu_probe(struct platform_device *pdev)
 	}
 
 	//firmware space for M0
-	dev->m0_p_fw_space_vir = ioremap(dev->m0_p_fw_space_phy,
+	dev->m0_p_fw_space_vir = ioremap_wc(dev->m0_p_fw_space_phy,
 			M0_BOOT_SIZE
 			);
 	if (!dev->m0_p_fw_space_vir)
@@ -2650,7 +2650,7 @@ static int vpu_probe(struct platform_device *pdev)
 
 	memset_io(dev->m0_p_fw_space_vir, 0, M0_BOOT_SIZE);
 
-	dev->m0_rpc_virt = ioremap(dev->m0_rpc_phy,
+	dev->m0_rpc_virt = ioremap_wc(dev->m0_rpc_phy,
 			SHARED_SIZE
 			);
 	if (!dev->m0_rpc_virt) {
diff --git a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
index ad9df20..2aab719 100644
--- a/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
+++ b/drivers/mxc/vpu-encoder-b0/vpu_encoder_b0.c
@@ -1442,7 +1442,7 @@ static int boot_CM4_up(struct vpu_dev *dev, void *boot_addr)
 	aux_core_ram = 0x34FE0000;
 	size = SZ_128K;
 
-	core_ram_vir = ioremap(aux_core_ram,
+	core_ram_vir = ioremap_wc(aux_core_ram,
 			size
 			);
 	if (!core_ram_vir)
@@ -1981,7 +1981,7 @@ static int vpu_probe(struct platform_device *pdev)
 		return ret;
 	}
 	//firmware space for M0
-	dev->m0_p_fw_space_vir = ioremap(dev->m0_p_fw_space_phy,
+	dev->m0_p_fw_space_vir = ioremap_wc(dev->m0_p_fw_space_phy,
 			M0_BOOT_SIZE
 			);
 	if (!dev->m0_p_fw_space_vir)
@@ -1989,7 +1989,7 @@ static int vpu_probe(struct platform_device *pdev)
 
 	memset_io(dev->m0_p_fw_space_vir, 0, M0_BOOT_SIZE);
 
-	dev->m0_rpc_virt = ioremap(dev->m0_rpc_phy,
+	dev->m0_rpc_virt = ioremap_wc(dev->m0_rpc_phy,
 			SHARED_SIZE
 			);
 	if (!dev->m0_rpc_virt)
-- 
1.7.9.5

